run

```diff
@@ -270,6 +292,28 @@ def build_swift_toolchain(workspace, args):
         raise common.UnsupportedPlatform
     common.check_execute(build_command, timeout=9999999)
 
+def build_clang(workspace, source_path):
+    build_path = os.path.join(workspace, 'build_clang_source_compat')
+    source_path = os.path.join(source_path, 'llvm')
+    common.check_execute(['mkdir', '-p', build_path])
+
+    with common.DirectoryContext(build_path):
+        try:
+            ninja_path = common.check_execute_output(['xcrun', '--find', 'ninja']).strip().decode("utf-8")
+            cmake_command = [
+                'xcrun', 'cmake', '-G', 'Ninja',
+                '-DCMAKE_MAKE_PROGRAM={}'.format(ninja_path),
+                '-DLLVM_ENABLE_PROJECTS=clang;llvm',
+                '-DCMAKE_BUILD_TYPE=Release',
+                '-DCLANG_APPLE_BUILD_VERSION_STRING=13000000',
+                '-DLLVM_TARGETS_TO_BUILD=X86;AArch64;ARM',
```

Comment:
No really blocking the PR but how did you come up with building for these architectures? IIUC we are building Swift for the host platform + `macosx-arm64` (based on https://github.com/apple/swift/blob/5e9ff1596061cfbde9c0ca0bc7497570914da6c3/utils/build-presets.ini#L2569).

I’m not entirely sure how this relates to the architectures you specified.

Reply:
Specifying possible compiler targets is different from setting cross-compilation hosts, the former is about what targets the compiler can compile for and the latter is about what host architecture the compiler executable itself runs on.

Hence, `DLLVM_TARGETS_TO_BUILD` is more related to `llvm-targets-to-build=X86;ARM;AArch64;PowerPC` in https://github.com/apple/swift/blob/5e9ff1596061cfbde9c0ca0bc7497570914da6c3/utils/build-presets.ini#L594

We just don't add `PowerPC` as we don't have adopters run on it.